src/analysis/processing/qgsalgorithmfieldcalculator.cpp

```diff
@@ -0,0 +1,223 @@
+}
+
+QString QgsFieldCalculatorAlgorithm::outputName() const
+{
+  return QObject::tr( "Calculated" );
+}
+
+QList<int> QgsFieldCalculatorAlgorithm::inputLayerTypes() const
+{
+  return QList<int>() << QgsProcessing::TypeVector;
+}
+
+QgsProcessingFeatureSource::Flag QgsFieldCalculatorAlgorithm::sourceFlags() const
+{
+  return QgsProcessingFeatureSource::FlagSkipGeometryValidityChecks;
+}
+
+void QgsFieldCalculatorAlgorithm::initParameters( const QVariantMap &configuration )
+{
+  Q_UNUSED( configuration );
+
+  QStringList fieldTypes = QStringList( {QObject::tr( "Float" ), QObject::tr( "Integer" ), QObject::tr( "String" ), QObject::tr( "Date" ) } );
+
+  std::unique_ptr< QgsProcessingParameterField > existingFieldName = qgis::make_unique< QgsProcessingParameterField > ( QStringLiteral( "EXISTING_FIELD_NAME" ), QObject::tr( "Result in existing field" ), QVariant(), QStringLiteral( "INPUT" ), QgsProcessingParameterField::Any, false, true );
+  std::unique_ptr< QgsProcessingParameterString > newFieldName = qgis::make_unique< QgsProcessingParameterString > ( QStringLiteral( "NEW_FIELD_NAME" ), QObject::tr( "Result in new field" ), QVariant(), false, true );
+  std::unique_ptr< QgsProcessingParameterEnum > fieldType = qgis::make_unique< QgsProcessingParameterEnum > ( QStringLiteral( "FIELD_TYPE" ), QObject::tr( "Result field type" ), fieldTypes, false, 0 );
+  std::unique_ptr< QgsProcessingParameterNumber > fieldLength = qgis::make_unique< QgsProcessingParameterNumber > ( QStringLiteral( "FIELD_LENGTH" ), QObject::tr( "Result field length" ), QgsProcessingParameterNumber::Integer, QVariant(0), false, 0 );
+  std::unique_ptr< QgsProcessingParameterNumber > fieldPrecision = qgis::make_unique< QgsProcessingParameterNumber > ( QStringLiteral( "FIELD_PRECISION" ), QObject::tr( "Result field precision" ), QgsProcessingParameterNumber::Integer, QVariant(0), false, 0 );
+  std::unique_ptr< QgsProcessingParameterExpression > expression = qgis::make_unique< QgsProcessingParameterExpression> ( QStringLiteral( "FORMULA" ), QObject::tr( "Formula" ), QVariant(), QStringLiteral( "INPUT" ), false );
+
+  expression->setMetadata( QVariantMap( {{"inlineEditor", true}} ) );
+
+  addParameter( existingFieldName.release() );
+  addParameter( newFieldName.release() );
+  addParameter( fieldType.release() );
+  addParameter( fieldLength.release() );
+  addParameter( fieldPrecision.release() );
+  addParameter( expression.release() );
+}
+
+QgsFields QgsFieldCalculatorAlgorithm::outputFields( const QgsFields & ) const
+{
+  return mFields;
+}
+
+QString QgsFieldCalculatorAlgorithm::shortHelpString() const
+{
+  return QObject::tr( "This algorithm computes a new vector layer with the same features of the input layer, "
+                      "but with an additional attribute. The values of this new attribute are computed from each "
+                      "feature using a mathematical formula, based on the properties and attributes of the feature."
```

Comment:
- Better talk about "an expression" instead of "a math formula"?

Reply:
It was copy paste from the python implementation. Nice isuggestion.